src/prefect/engine.py

```diff
@@ -2021,6 +2052,8 @@ async def set_state_and_handle_waits(set_state_func) -> OrchestrationResult:
     # Parse the response to return the new state
     if response.status == SetStateStatus.ACCEPT:
         # Update the state with the details if provided
+        state.id = response.state.id
```

Comment:
This part feels a bit... hacky. I considered changing it so that the `response.state` is returned, but we use the `state` for linking to results and that immediately started failing. I decided this was likely the least obtrusive change, but happy to consider alternatives.

Reply:
Yeah we want the client-side state so that we retain in-memory results. This lgtm.